__tests__/app.test.js

```diff
@@ -171,4 +171,55 @@ describe('GET /api/articles/:article_id/comments', () => {
         expect(comments).toEqual([])
       })
     });
+});
+
+describe('POST /api/articles/:article_id/comments', () => {
+  test('201: should respond with the posted comment', () => {
+    const newComment = {username: "butter_bridge", body: "First!"}
+    return request(app)
+    .post('/api/articles/2/comments')
+    .send(newComment)
+    .expect(201)
+    .then(({body}) => {
+      const {comment} = body;
+      expect(comment).toMatchObject({
+        comment_id: expect.any(Number),
+        body: "First!",
+        votes: 0,
+        author: "butter_bridge",
+        article_id: 2,
+        created_at: expect.any(String)
+      })
+    })
+  });
+  test('400: should respond with error when given invalid data type for article_id', () => {
+    const newComment = {username: "butter_bridge", body: "First!"}
+    return request(app)
+    .post('/api/articles/banana/comments')
+    .send(newComment)
+    .expect(400)
+    .then(({body}) => {
+      expect(body.msg).toBe("Bad request")
+    })
+  });
+  test('404: should respond with error when given non-existent article_id', () => {
+    const newComment = {username: "butter_bridge", body: "First!"}
+    return request(app)
+    .post('/api/articles/999/comments')
+    .send(newComment)
+    .expect(404)
+    .then(({body}) => {
+      expect(body.msg).toBe("Not found")
+    })
+  });
+  test('400: should respond with error when username/body are null/not provided', () => {
```

Comment:
Should add a test to provide a 404 if an invalid username provided
Other tests are great 👍